pontoon/base/models.py

```diff
@@ -377,6 +357,87 @@ def add_translation(self, string, locale, user):
 
         translation.save()
 
+    def get_translation(self, plural_form=None):
+        """Get fetched translation of a given entity."""
+        translations = self.fetched_translations
+
+        if translations:
+            if plural_form:
+                translations = [t for t in translations if t.plural_form == plural_form]
+
+            translation = sorted(translations, key=lambda k: (-k.approved, k.date))[0]
+
+            return {
+                'fuzzy': translation.fuzzy,
+                'string': translation.string,
+                'approved': translation.approved,
+                'pk': translation.pk
+            }
+
+        else:
+            return {
+                'fuzzy': False,
+                'string': u'',
+                'approved': False,
+                'pk': None
+            }
+
+    @classmethod
+    def serialize(self, project, locale, paths=None):
+        """Get project entities with locale translations."""
+
+        entities = self.objects.filter(resource__project=project, obsolete=False)
+
+        if paths:
+            entities = entities.filter(resource__path__in=paths)
+
+        entities = (entities
+                    .prefetch_related(
+                        Prefetch(
+                            'translation_set',
+                            queryset=Translation.objects.filter(locale=locale),
+                            to_attr='fetched_translations'
+                        )
+                    )
+                    .prefetch_related('resource')
+                    )
```

Comment:
Adding this on another line but not un-indented is weird. The hanging indent is useful for querysets where you're chaining multiple calls, but since we're just doing the one call to `prefetch_related` here, I'd recommend just indenting once:

```python
entities = entities.prefetch_related(
    'resource',
    Prefetch(
        # blah blah
    )
)
```